include/wil/registry_basic.h

```diff
@@ -77,1599 +77,1826 @@ namespace details
+                if (error == ERROR_FILE_NOT_FOUND && !::wil::reg::details::should_return_not_found<err_policy>())
+                {
+                    error = ERROR_SUCCESS;
+                    hkey.reset();
+                }
+                return err_policy::HResult(HRESULT_FROM_WIN32(error));
             }
 
-        private:
-            friend class reg_view_t;
-            // non-owning resource (just get a copy from the owner)
-            HKEY m_hkey;
-        };
-
-        struct value_enumerator
-        {
-            explicit value_enumerator(HKEY key) :
-                m_hkey(key)
+            ::wil::unique_hkey open_unique_key(_In_opt_ PCWSTR subKey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
             {
+                ::wil::unique_hkey local_key{};
+                open_key(subKey, local_key, access);
+                return local_key;
             }
 
-            // begin() will throw Registry::Exception on error 
-            [[nodiscard]] ::wil::registry::details::value_iterator begin() const
+#if defined(__WIL_WINREG_STL)
+            typename err_policy::result open_key(_In_opt_ PCWSTR subKey, ::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
             {
-                return ::wil::registry::details::value_iterator(m_hkey, ::wil::registry::details::iterator_creation_flag::begin);
+                auto error = ::RegOpenKeyExW(::wil::reg::details::get_key(m_key), subKey, 0, ::wil::reg::details::get_access_flags(access), ::wil::out_param(hkey));
+                if (error == ERROR_FILE_NOT_FOUND && !::wil::reg::details::should_return_not_found<err_policy>())
+                {
+                    error = ERROR_SUCCESS;
+                    hkey.reset();
+                }
+                return err_policy::HResult(HRESULT_FROM_WIN32(error));
             }
-
-            [[nodiscard]] ::wil::registry::details::value_iterator end() const WI_NOEXCEPT
+            ::wil::shared_hkey open_shared_key(_In_opt_ PCWSTR subKey, ::wil::reg::key_access access = ::wil::reg::key_access::read) const
             {
-                return ::wil::registry::details::value_iterator(m_hkey, ::wil::registry::details::iterator_creation_flag::end);
+                ::wil::shared_hkey local_key{};
+                open_key(subKey, local_key, access);
+                return local_key;
             }
+#endif // #if defined(__WIL_WINREG_STL)
 
-        private:
-            friend class reg_view_t;
-            // non-owning resource (just get a copy from the owner)
-            HKEY m_hkey;
-        };
+#if defined(__SDDL_H__)
+            typename err_policy::result create_key(_In_ PCWSTR subKey, ::wil::unique_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const

```

Comment:
Is there a good way to test these SDDL variants?

Reply:
yeah. I still need to add the create_key tests :(